spec/bagger_tag_spec.rb

```diff
@@ -0,0 +1,137 @@
+require "spec_helper"
+require "bagger_tag"
+
+describe BaggerTag do
+
+
+  describe "::from_hash" do
+    let(:json_tag) do
+      JSON.parse(<<JSON)
+      {
+        "SomeField": {
+          "fieldRequired": true,
+          "valueList": [
+            "allowed_value",
+            "another_allowed_value"
+          ]
+        }
+      }
+JSON
+
+  end
+    it "accepts a hash deserialized from a JSON bagger profile" do
+      expect(described_class.from_hash(json_tag)).not_to be(nil)
+    end
+
+    it "has a name that matches the json hash" do
+      expect(described_class.from_hash(json_tag).name).to eq("SomeField")
+    end
+
+    it "raises an exception if the provided hash has more than one key" do
+      expect do
+        described_class.from_hash(
+        { "SomeField" => { "fieldRequired" => true },
+          "AnotherField" => { "fieldRequired" => true } })
+      end.to raise_exception(ArgumentError)
+    end
+
+    it "raises an exception if the provided hash is empty" do
+      expect { described_class.from_hash({}) }.to raise_exception(ArgumentError)
+    end
+
+    it "raises an exception if the provided hash does not have a fieldRequired key" do
+      expect { described_class.from_hash({ "SomeField": {} }) }.to raise_exception(ArgumentError)
+    end
+
+  end
+
+  describe "#value_valid?" do
+    let(:errors) { [] }
+
+    shared_context "true with no errors with value" do |value|
+      it "is true" do
+        expect(subject.value_valid?(value)).to be true
+      end
+
+      it "does not report any errors" do
+        subject.value_valid?(value,errors: errors)
+        expect(errors).to be_empty
+      end
+    end
+
+    def expect_error_with_value(value,error_pattern)
+      subject.value_valid?(value,errors: errors)
+      expect(errors).to include a_string_matching error_pattern
+    end
+
+    context "with a required tag with no value list" do
+      subject { BaggerTag.new(name: "SomeField", required: true) }
+
+      it "is false when not given a value" do
```

Comment:
The topic and conditions for these `it` statements are in the describe/context and should just be "is false" and "reports an error".

Maybe change the context to "with a required tag but no value list".

Likewise where applicable for the following examples.

Reply:
The topic & conditions for the `it` statements are included in the context above (https://github.com/mlibrary/chipmunk/pull/33/files/b5762fc76e9e51ee40994d19690545c99d39dac3#diff-47c52f9c34a7dc40409bf0931d69b377R15) but I don't believe it's the case here. I agree the context names here could be approved. The context here is about the tag specification, the `it` is about the expected return & error when checking a given value against the tag specification.